harper-core/src/spell/full_dictionary.rs

```diff
@@ -103,11 +113,13 @@ impl FullDictionary {
     /// list. NOTE: This function will sort the original word list by its
     /// length. If the word list's order is changed after creating the
     /// lookup, it will no longer be valid.
-    fn create_len_starts(words: &mut [CharString]) -> Vec<usize> {
-        words.sort_by_key(|a| a.len());
+    fn create_len_starts(words: &[CharString]) -> Vec<usize> {
```

Comment:
I'm not directly opposed to making the argument here immutable, I'm just curious your reasoning.

Reply:
I feel like it just makes the logic of sorting by length more explicit in the constructor. I had trouble realizing that was occuring because I expected this function not to have side-effects.
This is especially important because if the way that the words array is sorted changes in any way, it needs to match the sorting when creating the precomputed map.

Reply:
Alright. Well, if you change the behavior of the function, make sure you update that function's comment (which claims the function mutates the provided word list).